stackslib/src/net/download/epoch2x.rs

```diff
@@ -56,22 +56,22 @@ use crate::net::{
 };
 use crate::util_lib::db::{DBConn, Error as db_error};
 
-#[cfg(not(test))]
+#[cfg(not(any(test, feature = "testing")))]
 pub const BLOCK_DOWNLOAD_INTERVAL: u64 = 180;
-#[cfg(test)]
+#[cfg(any(test, feature = "testing"))]
 pub const BLOCK_DOWNLOAD_INTERVAL: u64 = 0;
 
 /// If a URL never connects, don't use it again for this many seconds
-#[cfg(not(test))]
+#[cfg(not(any(test, feature = "testing")))]
 pub const BLOCK_DOWNLOAD_BAN_URL: u64 = 300;
-#[cfg(test)]
+#[cfg(any(test, feature = "testing"))]
 pub const BLOCK_DOWNLOAD_BAN_URL: u64 = 60;
 
 /// If we created a request to download a block or microblock, don't do so again until this many
 /// seconds have passed.
-#[cfg(not(test))]
+#[cfg(not(any(test, feature = "testing")))]
 pub const BLOCK_REREQUEST_INTERVAL: u64 = 60;
-#[cfg(test)]
+#[cfg(any(test, feature = "testing"))]
 pub const BLOCK_REREQUEST_INTERVAL: u64 = 30;
```

Comment:
These are kind of scary to me -- the `testing` feature is a default feature for `debug` builds (not just test builds), so changing the values of constants for all `debug` builds could have some unintended consequences.

Reply:
>  the testing feature is a default feature for debug builds (not just test builds),

I did not know that.  That sounds like a bug -- `"testing"` is supposed to activate test-only code paths in dependent crates, I thought?

Reply:
Can you show me where `"testing"` is an enabled feature for debug builds?  Because, AFAIK that was never supposed to be the case.

Reply:
It's codepaths intended for running only in tests, but cargo doesn't have an option for "enable feature only in testing" (see maybe this issue: https://github.com/rust-lang/cargo/issues/8379), so the workaround is to enable it by default for debug builds. I totally agree that its a footgun, though, and I think we should try to deprecate the `testing` feature.

Reply:
> Can you show me where "testing" is an enabled feature for debug builds? Because, AFAIK that was never supposed to be the case.

https://github.com/stacks-network/stacks-core/blob/master/testnet/stacks-node/Cargo.toml#L43-L45

It's basically they only way to invoke "test" code from a separate crate's tests for the reasons I mentioned above.

Reply:
Ah -- nevermind, you are right. `[dev-dependencies]` only gets used for test builds, examples, docs, and benchmarks. I assumed that `profile.dev` builds also used them (probably because of the name?) but that's not the case. After applying this patch:

```git
diff --git a/testnet/stacks-node/src/main.rs b/testnet/stacks-node/src/main.rs
index fcdc9f584..b55006b56 100644
--- a/testnet/stacks-node/src/main.rs
+++ b/testnet/stacks-node/src/main.rs
@@ -281,6 +281,10 @@ fn main() {
 
     let mut args = Arguments::from_env();
     let subcommand = args.subcommand().unwrap().unwrap_or_default();
+    if subcommand.as_str() == "test-build" {
+        info!("build: {}. block_request_int = {}", version(), stacks::net::download::epoch2x::BLOCK_REREQUEST_INTERVAL);
+        return;
+    }
 
     info!("{}", version());
```

And running:
```
% cargo run --bin=stacks-node -- test-build  
   ...
    Finished `dev` profile [optimized + debuginfo] target(s) in 0.12s
     Running `target/debug/stacks-node test-build`
INFO [1726508395.509174] [testnet/stacks-node/src/main.rs:285] [main] build: stacks-node 0.1.0 (:, debug build, macos [aarch64]). block_request_int = 60
```